### PR TITLE
Add Remove Shared Access feature

### PR DESCRIPTION
**Pull request by scouillard**
## Description

Users should be able to delete their own Shared Access

## Testing Steps

Some things to consider & test:

- A user can delete its own share access. In that case, the user is redirected to /rooms
- A user can delete its own share access AND be an admin with ManageUsers. In that case, the admin is not redirected to /rooms

## Screenshots (if appropriate):

image

### CODE REVIEW

File: app/services/permissions_checker.rb [hunk 1]
```diff
@@ -61,7 +62,8 @@ def authorize_manage_rooms
   def authorize_shared_room
     return false if @friendly_id.blank? && @record_id.blank?
 
-    @current_user.shared_rooms.exists?(friendly_id: @friendly_id) ||
+    SharedAccess.joins(:shared_room).find_by(user_id: @current_user.id, 'shared_room.friendly_id': @friendly_id).present? ||
```
Comment (farhatahmad) [inline app/services/permissions_checker.rb:65]: Why the change here?

**Comment by farhatahmad** [comment]
Question - wouldn't this be much simpler if we just created another endpoint?

File: app/services/permissions_checker.rb (hunk 1 above)
Comment (scouillard) [reply app/services/permissions_checker.rb:65]: Doesn't this solve the "delete another shared access" issue actually?
It checks if the user is trying to do an action on the SharedAccess specifically, instead of just checking if the room is a shared room of the user.

File: app/javascript/components/rooms/room/FeatureTabs.jsx [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
+import useRoom from '../../../hooks/queries/rooms/useRoom';
 
-export default function FeatureTabs({ shared }) {
+export default function FeatureTabs() {
   const { t } = useTranslation();
   const { isLoading, data: settings } = useSiteSetting(['PreuploadPresentation', 'ShareRooms']);
   const currentUser = useAuth();
 
+  const { friendlyId } = useParams();
+  const { data: room } = useRoom(friendlyId);
```
Comment (scouillard) [inline app/javascript/components/rooms/room/FeatureTabs.jsx:19]: This feels not optimised.
We are querying the Room inside every Feature tab - it could be passed down from the Room component.
Not sure how React Query works - maybe it is not too bad.
But it is either that or a (somewhat) major refactoring with placeholders etc.

File: app/javascript/components/rooms/room/FeatureTabs.jsx [hunk 3]
```diff
@@ -27,17 +31,6 @@ export default function FeatureTabs({ shared }) {
...
-  // Returns only the Recording tab if the room is a Shared Room and the User does not have the ManageRoom permission
-  if (shared && !currentUser?.permissions?.ManageRooms) {
-    return (
-      <Tabs className="wide-white pt-4 mx-0" defaultActiveKey="recordings" unmountOnExit>
-        <Tab className="background-whitesmoke" eventKey="recordings" title={t('recording.recording')}>
-          <RoomRecordings />
-        </Tab>
-      </Tabs>
-    );
-  }
```
Comment (scouillard) [inline app/javascript/components/rooms/room/FeatureTabs.jsx:39]: This has never worked, related to the comment above.
Passed down props was not properly dealt with - it needs a placeholder/isLoading.
It was set by default as 'false' and would not re-render for some reason.

File: app/javascript/components/rooms/room/FeatureTabs.jsx [hunk 4]
```diff
@@ -61,11 +54,3 @@ export default function FeatureTabs({ shared }) {
...
   );
 }
-
-FeatureTabs.defaultProps = {
-  shared: false,
-};
-
-FeatureTabs.propTypes = {
-  shared: PropTypes.bool,
-};
```
Comment (scouillard) [inline app/javascript/components/rooms/room/FeatureTabs.jsx:71]: Related to the other comments

**Review by farhatahmad — DISMISSED** [review]

File: app/controllers/api/v1/shared_accesses_controller.rb [hunk 5]
```diff
@@ -38,6 +38,21 @@ def destroy
...
+
+        render_data status: :ok
+      end
+
+      # GET /api/v1/shared_accesses/:friendly_id.json
+      # Returns a list of all of the room's shared users
+      def show
+        render_data data: @room.shared_users.search(params[:search]), serializer: SharedAccessSerializer, status: :ok
+      end
+
```
Comment (farhatahmad) [inline app/controllers/api/v1/shared_accesses_controller.rb:55]: Duplicated code?

**Comment by farhatahmad** [comment]
@scouillard Failing rubocop

**Review by farhatahmad — APPROVED** [review]
